ludwig/models/llm.py

```diff
@@ -451,6 +451,21 @@ def generate(
 
         return outputs
 
+    def merge_and_unload(self, progressbar: bool = False) -> None:
+        """This method merges the LoRa layers into the base model.  This is needed if someone wants to use the base
+        model as a standalone model.  The implementation calls merge_and_unload() of the underlying LoraModel class
+        (in peft).
+
+        Args:
+            progressbar (bool): whether to show a progressbar indicating the unload and merge process
+        """
+        from peft import LoraModel
+
+        if isinstance(self.model.base_model, LoraModel):
+            self.model.base_model.merge_and_unload(progressbar=progressbar)
```

Comment:
Nice! In a follow-up PR, I think it could be also good to add support for AdaLoRAModel since I believe it also allows `merge_and_unload`

Reply:
Oh good catch, you're right :) Didn't realize it was a subclass so makes sense. Thank you!

Reply:
🙇